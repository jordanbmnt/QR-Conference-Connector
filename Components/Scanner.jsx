import { StyleSheet } from "react-native";
import { Box, Heading } from "@gluestack-ui/themed";
import { BarCodeScanner } from "expo-barcode-scanner";

export default Scanner = ({ scanCheck, handleScanned }) => {
  return (
    <Box sx={styles.codeScanner}>
      <Heading sx={styles.qRHeading} size='3xl'>
        Scan QR
      </Heading>
      <BarCodeScanner
        onBarCodeScanned={scanCheck ? undefined : handleScanned}
        style={[StyleSheet.absoluteFillObject, { marginBottom: 0 }]}
      />
    </Box>
  );
};

const styles = StyleSheet.create({
  qRHeading: {
    color: "$textLight100",
    padding: "$1.5",
    backgroundColor: "#131721",
    width: "100%",
    height: "16%",
    textAlign: "center",
    paddingTop: "$8",
    borderRadius: 9,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    zIndex: 1,
  },
  codeScanner: { width: "100%", height: "90%", paddingBottom: 0 },
});
