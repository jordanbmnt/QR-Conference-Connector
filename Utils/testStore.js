import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { setupUserInfoStore } from "../Redux/store";
import { GluestackUIStyledProvider } from "@gluestack-ui/themed";
import { config } from "@gluestack-ui/config";

export const initialState = {
  "appSlice": {
    viewContactInfo: false,
  scanViewVisible: true,
  previousPage: false,
  contactInfo: {},
  alertInfo: {
    visibility: false,
    theme: "",
  }
  },
};

export function renderWithProviders(
  ui,
  {
    preloadedState = initialState,
    store = setupUserInfoStore(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <GluestackUIStyledProvider config={config}>
          {children}
        </GluestackUIStyledProvider>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
