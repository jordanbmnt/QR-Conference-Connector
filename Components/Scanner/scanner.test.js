import Scanner from "./Scanner";
import { renderWithProviders } from "../../Utils/testStore";

describe("Scanner", function () {
  let scanner;
  beforeEach(() => {
    scanner = renderWithProviders(
      <Scanner scanCheck={false} handleScanned={"handleBarCodeScanned"} />
    );
  });

  it("should display a heading of 'Scan QR'", function () {
    const scanQRCount = scanner.getAllByText("Scan QR").length;
    expect(scanQRCount).toBe(1);
  });
  it("should render the Scanner component correctly", () => {
    expect(scanner).toMatchSnapshot();
  });
});
