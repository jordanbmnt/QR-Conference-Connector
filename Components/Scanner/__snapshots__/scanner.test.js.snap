// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scanner should render the Scanner component correctly 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {},
      undefined,
      {
        "height": "90%",
        "paddingBottom": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
  sx={
    {
      "height": "90%",
      "paddingBottom": 0,
      "width": "100%",
    }
  }
>
  <Text
    accessibilityRole="header"
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      {
        "backgroundColor": "#131721",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderRadius": 9,
        "color": "#E5E5E5",
        "fontFamily": undefined,
        "fontSize": 36,
        "fontWeight": "700",
        "height": "16%",
        "letterSpacing": -0.2,
        "lineHeight": 56,
        "marginVertical": 0,
        "padding": 6,
        "paddingTop": 32,
        "textAlign": "center",
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    Scan QR
  </Text>
  <ViewManagerAdapter_ExpoBarCodeScanner
    barCodeTypes={[]}
    onBarCodeScanned={[Function]}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "marginBottom": 0,
        },
      ]
    }
  />
</View>
`;
